addons/onboarding/tools/progress_step.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from enum import Enum
+
+
+class StepValidationResponse(Enum):
```

Comment:
I don't really see the added value of using an Enum vs just using string as response to be honest.

Especially considering that the webclient will not be able to use an Enum anyway and will have to rely on string values, it just adds some ping pong to check what is returned by the various actions.

Enum does not seem vastly used within Odoo, I only see one other extension.

Reply:
The added value is that when one see a `StepValidationResponse.NOT_FOUND`, one knows where to look for any other responses that could be used.

If we used "magic" strings "NOT_FOUND", "WAS_DONE", without other reference, it's difficult to identify all the possible return values.
It is true that the webclient cannot handle these, but the server method giving the value will have that reference.

This comes from having had to understand magic strings myself and I don't wish it for anyone ;-)

Reply:
Well the problem with that is that python does not enforce types so nothing will force people to use the Enum constants instead of the magic strings.

To prove my point, it seems that even you in your enterprise PR don't use the Enum:
![image](https://user-images.githubusercontent.com/41282536/229703920-ab36d77a-a490-4db0-842d-4d5c29c844de.png)

The Enum values are exactly matching the strings anyway, so it's not like there is a lot of magic attached to them.

Still not convinced it's useful, and I don't like to introduce new patterns in the Odoo codebase if I don't see a real added value.
But it's a bit a matter of taste, so I suppose I don't have any very strong argument against it.
If you really really want to keep them, we can keep them (but at least use them everywhere possible then).

Reply:
The actions return the string as it is the value that the webclient should receive, this is what we test here.
That said, as I also added the common method for validating steps from xmlid, users shouldn't have to return values "on their own", all should come from there. 

I'll drop the Enum.

Reply:
This makes me sad :(